File: internal/indexer/testdata/js/services/userService.js
const MAX_WORKERS = 4;

function processUserData(userId, data) {
    return Object.keys(data).length > 0;
}

function validateEmail(email) {
    return email.includes("@");
}

const formatUserName = (first, last) => {
    return `${first.trim()} ${last.trim()}`;
};

class UserRepository {
    findById(userId) {
        return { id: String(userId) };
    }

    save(user) {
        return true;
    }
}

module.exports = { processUserData, validateEmail, formatUserName, UserRepository };

File: internal/indexer/testdata/rust/services/user_service.rs
pub const MAX_WORKERS: u32 = 4;

pub fn process_user_data(user_id: u64, data: &str) -> bool {
    !data.is_empty()
}

pub fn validate_email(email: &str) -> bool {
    email.contains('@')
}

pub fn format_user_name(first: &str, last: &str) -> String {
    format!("{} {}", first.trim(), last.trim())
}

pub struct UserRepository;

impl UserRepository {
    pub fn find_by_id(&self, user_id: u64) -> Option<String> {
        Some(format!("user-{}", user_id))
    }

    pub fn save(&self, user: &str) -> bool {
        !user.is_empty()
    }
}
